veg2hab/vegkartering.py

```diff
@@ -446,27 +470,33 @@ def build_aggregate_habtype_field(row: gpd.GeoSeries) -> str:
     vegtypeinfos = row["VegTypeInfo"]
 
     assert len(habitatkeuzes) > 0, "Er moet minstens 1 habitatkeuze zijn"
+    assert len(habitatkeuzes) == len(
+        vegtypeinfos
+    ), "Er moet voor iedere habitatkeuze een VegTypeInfo zijn"
 
     # Hierin krijgen we per (habtype, kwaliteit) tuple de som van de percentages
     aggregate = defaultdict(float)
 
-    # Als het vlak geen opgegeven habitattype heeft, heeft het geen vegtype infos,
-    # dus heeft het ook geen opgegeven percentage, dus moeten we die handmatig op 100 zetten
-    if habitatkeuzes[0].status == KeuzeStatus.GEEN_OPGEGEVEN_VEGTYPEN:
-        assert (
-            len(habitatkeuzes) == 1
-        ), "Bij KeuzeStatus GEEN_OPGEGEVEN_VEGTYPE mag er maar 1 habitatkeuze zijn"
-        assert vegtypeinfos == [
-            VegTypeInfo(percentage=100, SBB=[], VvN=[])
-        ], "Bij KeuzeStatus GEEN_OPGEGEVEN_VEGTYPE moet er een leeg 100% VegTypeInfo zijn"
-        aggregate[
-            (habitatkeuzes[0].habtype, habitatkeuzes[0].kwaliteit.as_letter())
-        ] = 100
-    else:
-        # In alle andere gevallen kunnen we gewoon de percentages bij de habitatkeuze
-        # horende VegTypeInfos gebuiken
-        for keuze, info in zip(habitatkeuzes, vegtypeinfos):
-            aggregate[(keuze.habtype, keuze.kwaliteit.as_letter())] += info.percentage
+    # # Als het vlak geen opgegeven habitattype heeft, heeft het geen vegtype infos,
+    # # dus heeft het ook geen opgegeven percentage, dus moeten we die handmatig op 100 zetten
+    # if len(habitatkeuzes) == 1 and habitatkeuzes[0].status == KeuzeStatus.GEEN_OPGEGEVEN_VEGTYPEN:
+    #     if len(habitatkeuzes) > 1:
+    #         print("hiero")
+    #     assert (
+    #         len(habitatkeuzes) == 1
+    #     ), "Bij KeuzeStatus GEEN_OPGEGEVEN_VEGTYPE mag er maar 1 habitatkeuze zijn"
+    #     assert vegtypeinfos == [
+    #         VegTypeInfo(percentage=100, SBB=[], VvN=[])
+    #     ], "Bij KeuzeStatus GEEN_OPGEGEVEN_VEGTYPE moet er een leeg 100% VegTypeInfo zijn"
+    #     aggregate[
+    #         (habitatkeuzes[0].habtype, habitatkeuzes[0].kwaliteit)
+    #     ] = 100
+    # else:
+    #     # In alle andere gevallen kunnen we gewoon de percentages bij de habitatkeuze
+    #     # horende VegTypeInfos gebuiken
+
+    for keuze, info in zip(habitatkeuzes, vegtypeinfos):
+        aggregate[(keuze.habtype, keuze.kwaliteit)] += info.percentage
```

Comment:
Ik voegde hier kwaliteit.as_letter toe, en dan checkte ik er onder of kwaliteit een enum waarde was, dus hij liet printte nooit de kwaliteit.... Maar nu is t gefixt